tests/FSharp.Compiler.ComponentTests/Conformance/BasicGrammarElements/UseBindings/UseBang01.fs

```diff
@@ -0,0 +1,58 @@
+﻿open System
+
+type Disposable() =
+    [<DefaultValue>] static val mutable private disposedTimes: int
+    [<DefaultValue>] static val mutable private constructedTimes: int
+
+    do Disposable.constructedTimes <- Disposable.constructedTimes + 1
+
+    static member DisposeCallCount() = Disposable.disposedTimes
+    static member ConstructorCallCount() = Disposable.constructedTimes
+
+    interface IDisposable with
+        member _.Dispose() =
+            Disposable.disposedTimes <- Disposable.disposedTimes + 1
+
+type DisposableBuilder() =
+    member _.Using(resource: #IDisposable, f) =
+        async {
+            use res = resource
+            return! f res
+        }
+        
+    member _.Bind(disposable: Disposable, f) =
+        async {
+            let c = disposable
+            return! f c
+        }
+    
+    member _.Return(x) = async.Return x
+    
+    member _.ReturnFrom(x) = x
+    
+    member _.Bind(task, f) = async.Bind(task, f)
+
+let counterDisposable = DisposableBuilder()
+
+let example() =
+    counterDisposable {
+        use! res = new Disposable()
+        use! __ = new Disposable()
+        use! (res1) = new Disposable()
+        use! (___) = new Disposable()
+        use! _ = new Disposable()
```

Comment:
Other test dimensions:


- `use!` the exact same pattern, plain `_`, multiple times

AND:
- Assert not only the right amount if disposals, but also that each object was disposed (and none was disposed twice).

(i.e. test that there is no mistake when same pattern is used twice, e.g. `_`, and first gets disposed n-times and remaining ones do not get disposed at all)

Reply:
You could e.g. pass unique identifiers to each constructor of the `Disposable(..)` here, and then track disposal of each.

Reply:
See https://github.com/dotnet/fsharp/pull/18487/commits/d1532fe7ab528d47f3152080ff1d22d6346aaa54